src/pop2net/location.py

```diff
@@ -34,22 +34,25 @@ def agents(self) -> AgentList:
         """
         return self.model.agents_of_location(self)
 
-    def add_agent(self, agent: _agent.Agent) -> None:
+    def add_agent(self, agent: _agent.Agent, weight: float | None = None) -> None:
         """Assigns the given agent to this location.
 
         Args:
             agent: The agent that should be added to the location.
+            weight: The edge weight between the agent and the location. Defaults to 1.
         """
-        self.model.add_agent_to_location(self, agent)
+        self.model.add_agent_to_location(self, agent=agent, weight=weight)
 
-    def add_agents(self, agents: list) -> None:
+    def add_agents(self, agents: list, weight: float | None = None) -> None:
         """Add multiple agents at once.
 
         Args:
             agents (list): An iterable over agents.
+            weight(float | None): The edge weight between the agents and the location.
```

Comment:
If weight is set to None, the weight is generated by location.weight(), which returns 1 by default. So the users can either set a weight explicitly here by setting the argument weight or let the location generate (inidividual) weights.

Reply:
Might be a dumb question but why is the default not just 1 instead?

Reply:
Because `None` stands for "No I don't want to set the weight here. Just let the location set the weight by using location.weight(agent)". And location.weight(agent) could also return something different than 1. To use location.weight() is the preferred way and the more pop2net-way of setting the weight, so usually the users should let the weight argument here untouched. But I also wanted to give them the possibility to set the weight explicitly here.

Reply:
ah ok.